RUFAS/biophysical/animal/pen.py

```diff
@@ -933,11 +933,16 @@ def formulate_optimized_ration(  # noqa: C901
 
         previous_ration = getattr(self, "ration", None)
         num_attempts = 0
+        self.set_animal_nutritional_requirements(temperature=temperature, available_feeds=pen_available_feeds)
+        initial_dry_matter_requirement = self.average_nutrition_requirements.dry_matter
+        initial_protein_requirement = self.average_nutrition_requirements.metabolizable_protein
```

Comment:
```suggestion
        self.set_animal_nutritional_requirements(temperature=temperature, available_feeds=pen_available_feeds)
        initial_pen_average_nutrition_requirements = self.average_nutrition_requirements
        initial_dry_matter_requirement = initial_pen_average_nutrition_requirements.dry_matter
        initial_protein_requirement = initial_pen_average_nutrition_requirements.metabolizable_protein
```
So that we don't have to calculate the pen average nutrition requirements twice.